sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -282,7 +282,7 @@ describeColName
     ;
 
 ctes
-    : WITH namedQuery (',' namedQuery)*
+    : WITH recursive=RECURSIVE? namedQuery (',' namedQuery)*
```

Comment:
Seems that it is not required to have `RECURSIVE` in the query in some databases for recursive query?